src/components/MainSidebar/ProgramDimensionsPanel/ProgramDimensionsPanel.js

```diff
@@ -101,17 +105,22 @@ const ProgramDimensionsPanel = ({ visible }) => {
         setDimensionType(DIMENSION_TYPE_ALL)
     }, [inputType])
 
+    useEffect(() => {
+        // Clear everything but program itself when user switches program
+        dispatch(acClearUiStage())
+        setSearchTerm('')
+        setDimensionType(DIMENSION_TYPE_ALL)
+    }, [inputType, selectedProgramId])
+
     useEffect(() => {
         if (
             // These only have a single artificial stage
             inputType === OUTPUT_TYPE_EVENT &&
             programType === PROGRAM_TYPE_WITHOUT_REGISTRATION
         ) {
-            const artificialStageId = selectedProgram.programStages[0].id
-            const metadata = {
-                [artificialStageId]: selectedProgram.programStages[0],
-            }
-            dispatch(acUpdateUiProgramStageId(artificialStageId, metadata))
+            const artificialStage = selectedProgram.programStages[0]
+            const metadata = { [artificialStage.id]: artificialStage }
+            dispatch(acUpdateUiProgramStage(artificialStage.id, metadata))
```

Comment:
This has been renamed
```suggestion
            dispatch(acUpdateUiProgramStageId(artificialStage.id, metadata))
```